config/application.rb

```diff
@@ -151,6 +151,14 @@ class Application < Rails::Application
       Vmdb::Loggers.apply_config(::Settings.log)
     end
 
+    # register plugins even before loading settings, as plugins can bring their own settings
+    initializer :register_vmdb_plugins, :before => :load_vmdb_settings do
+      Rails.application.railties.each do |railtie|
+        next unless railtie.class.name.start_with?("ManageIQ::Providers::") || railtie.try(:vmdb_plugin?)
+        Vmdb::Plugins.instance.register_vmdb_plugin(railtie)
+      end
+    end
+
```

Comment:
Sure. Makes sense. Somehow I thought it has to be defined _after_ because of the mention in the `:before` keyword. But it's just symbols...